sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -166,6 +166,30 @@ object TypeCoercion {
     case (l, r) => None
   }
 
+  private def mergeComplexTypes(
+      t1: DataType,
+      t2: DataType,
```

Comment:
Maybe `findTightestCommonType` can also reuse `mergeComplexTypes` ?

Reply:
Seems like the behavior will be changed if we reuse there, and the test [TypeCoercionSuite.scala#L397-L400](https://github.com/apache/spark/blob/bed6849dbf51e1981772cd353ce1a7ae4f0626e2/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercionSuite.scala#L397-L400) fails.

Reply:
The test result might be wrong?

Reply:
yea it's wrong after this patch